source/exe/posix/legacy_main.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include "server/options_impl.h"
+
+namespace Envoy {
+
+/**
+ * Legacy implementation of main_common.
```

Comment:
@jmarantz should we just nuke this in this PR and have Envoy import on-call fix up the Google-side code?

Reply:
Actually I think there are no callers in Google to main_common(). Let's just nuke. I can send a separate PR if that helps.

Reply:
https://github.com/envoyproxy/envoy/pull/5298